repo_validation/scripts/validate_no_tabs.ps1

```diff
@@ -0,0 +1,194 @@
+Write-Host ""
+
+foreach ($file in $allFiles) {
+    # Check if file should be excluded
+    $relativePath = $file.FullName.Substring($RepoRoot.Length).TrimStart('\', '/')
+    $isExcluded = $false
+    
+    foreach ($excludeDir in $excludeDirs) {
+        if ($relativePath -like "$excludeDir\*" -or $relativePath -like "$excludeDir/*") {
+            $isExcluded = $true
+            $skippedFiles++
+            break
+        }
+    }
+    
+    if ($isExcluded) {
+        continue
+    }
+    
+    $totalFiles++
+    
+    # Read file content as raw text to preserve encoding
+    try {
+        $content = Get-Content -Path $file.FullName -Raw -ErrorAction Stop
+    }
+    catch {
+        Write-Host "  [WARN] Cannot read file: $($file.FullName)" -ForegroundColor Yellow
+        continue
+    }
+    
+    # Check if file contains tabs (ASCII 9)
+    if ($content -match "`t") {
+        # Count tabs for reporting
+        $tabCount = ([regex]::Matches($content, "`t")).Count
+        
+        Write-Host "  [FAIL] $($file.FullName)" -ForegroundColor Red
+        Write-Host "         Contains $tabCount tab character(s)" -ForegroundColor Yellow
+        
+        $filesWithTabs += [PSCustomObject]@{
+            FilePath = $file.FullName
+            TabCount = $tabCount
+        }
+        
+        if ($Fix) {
+            try {
+                # Replace all tabs with 4 spaces
+                $fixedContent = $content -replace "`t", "    "
+                
+                # Write back to file, preserving encoding
+                # Use UTF8 without BOM for consistency
+                $utf8NoBom = New-Object System.Text.UTF8Encoding $false
+                [System.IO.File]::WriteAllText($file.FullName, $fixedContent, $utf8NoBom)
+                
+                Write-Host "         [FIXED] Replaced $tabCount tab(s) with spaces" -ForegroundColor Green
+                $fixedFiles += $file.FullName
+            }
+            catch {
+                Write-Host "         [ERROR] Failed to fix: $_" -ForegroundColor Red
+            }
+        }
+    }
+}
+
+Write-Host ""
+Write-Host "========================================" -ForegroundColor Cyan
+Write-Host "Validation Summary" -ForegroundColor Cyan
+Write-Host "========================================" -ForegroundColor Cyan
+Write-Host "Total source files checked: $totalFiles" -ForegroundColor White
+Write-Host "Files skipped (excluded directories): $skippedFiles" -ForegroundColor White
+
+if ($Fix -and $fixedFiles.Count -gt 0) {
+    Write-Host "Files fixed successfully: $($fixedFiles.Count)" -ForegroundColor Green
+}
+
+if ($filesWithTabs.Count -gt 0 -and -not $Fix) {
+    Write-Host "Files with tabs: $($filesWithTabs.Count)" -ForegroundColor Red
+    Write-Host ""
+    Write-Host "The following files contain tab characters:" -ForegroundColor Yellow
+    
+    # Show first 50 files
+    $displayCount = [Math]::Min(50, $filesWithTabs.Count)
```

Comment:
interesting that we only print 50 in this file (and no such check in validate_file_endings)...